### ISSUE TITLE
Cast exception after upgrading to 6.0

### ISSUE THREAD

**Issue by mteper** [reactiveui/ReactiveUI#667]
In Xam.Mac project, binding a command results in cast exception.

Binding code:
`this.BindCommand(ViewModel, vm => vm.AcceptSelectedCommand, controller => controller.AcceptSelectedButton);`

Exception:
Cannot cast from source type to destination type.

```
  at (wrapper castclass) object:__castclass_with_cache (object,intptr,intptr)
  at ReactiveUI.ReactiveNotifyPropertyChangedMixin+<SubscribeToExpressionChain>c__AnonStorey3`2[XTractor.UI.ViewModels.ReviewStringsTaskViewModel,System.IObservable`1[ReactiveUI.IObservedChange`2[XTractor.UI.Models.SelectableNode,System.Object]]].<>m__1 (IObservedChange`2 x) [0x00000] in /Users/paul/code/reactiveui/ReactiveUI/ReactiveUI/ReactiveNotifyPropertyChangedMixin.cs:113 
  at System.Reactive.Linq.ObservableImpl.Select`2+_[ReactiveUI.IObservedChange`2[System.Object,System.Object],ReactiveUI.ObservedChange`2[XTractor.UI.ViewModels.ReviewStringsTaskViewModel,System.IObservable`1[ReactiveUI.IObservedChange`2[XTractor.UI.Models.SelectableNode,System.Object]]]].OnNext (IObservedChange`2 value) [0x00000] in <filename unknown>:0 
```

**Comment by kentcb** [comment]
@mteper any chance you could test this with the new RxCommand stuff coming in 7? PR is #971.

**Comment by mteper** [comment]
This was actually fixed back in 6.5.x timeframe.
